File: Assets/Scripts/LoSTrigger.js
#pragma strict
var losTarget : GameObject;
public var triggered : boolean;

function Awake (){
	losTarget = GameObject.Find("EmptyTarget");
}


function OnTriggerEnter (collision : Collider) {

	if (collision){
		if(collision.gameObject.tag == "Player"){

			losTarget = collision.gameObject;
			triggered = true;
		}
	}
}

File: Assets/Scripts/LosTrigger.cs
using UnityEngine;
using System.Collections;

public class LosTrigger : MonoBehaviour {

	public GameObject losTarget;
	public bool triggered;
	public bool onSightArea;

	void OnTriggerStay (Collider col) {
		
		if(col.gameObject.tag == "Player"){
			losTarget = col.gameObject;
			onSightArea = true;
			triggered = true;
		}
	}

	void OnTriggerExit (Collider col){

		if (col.gameObject.tag == "Player") {
			onSightArea = false;
		}
	}
}
